lib/apartment/adapters/sqlite3_adapter.rb

```diff
@@ -13,6 +13,9 @@ def initialize(config)
         @default_dir = File.expand_path(File.dirname(config[:database]))
 
         super
+
+        Apartment.connection_handler = ActiveRecord::ConnectionAdapters::ConnectionHandler.new
+        reset
```

Comment:
Ya I suppose. I'm wary of deeply nested inheritance as it often results in messy code if the abstraction is wrong. IMO its better to duplicate first, then see where the similarities are as you continue to modify adapters etc in order to develop the right abstraction. I'm not confident *yet* that a `connection based adapter` is necessarily the correct abstraction